### PR TITLE
Issue #3177657 by tBKoT: Add posibility to filter KPI by terms.

### PR DESCRIPTION
**Pull request by dmytriikaiun**
**Related links**:
- https://www.drupal.org/project/kpi_analytics/issues/3177657
- https://getopensocial.atlassian.net/browse/YANG-3595
- https://getopensocial.atlassian.net/browse/YANG-3596

### CODE REVIEW

File: assets/js/morris-chart.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   Drupal.behaviors.kpiAnalyticsRenderMorris = {
     attach: function (context, settings) {
-      $(context).find('div.morris_chart').once('renderChart').each(function () {
+      $('div.morris_chart').once('renderChart').each(function () {
```
Comment (ribel) [inline assets/js/morris-chart.js:7]: @dmytriikaiun could you explain why do you make this change?

File: src/Plugin/Block/KPIBlockContentBlock.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        '#value' => $default_vocabulary,
+        '#ajax' => [
+          'event' => 'change',
+          'callback' => [$this, 'updateTermList'],
+          'wrapper' => 'kpi-term-wrapper',
+        ],
+        '#required' => TRUE,
+      ],
+      'terms' => [
+        '#type' => 'select2',
```
Comment (ribel) [inline src/Plugin/Block/KPIBlockContentBlock.php:64]: @dmytriikaiun I think we need to add a dependency from select2 in this case.

File: src/Plugin/KPIDatasource/DrupalKPIDatasource.php [hunk 3]
```diff
@@ -20,16 +16,21 @@
 class DrupalKPIDatasource extends KPIDatasourceBase {
 
   /**
-   * @inheritdoc
+   * {@inheritdoc}
    */
-  public function query($query) {
+  public function query(BlockContentInterface $entity, $block) {
     $data = [];
     // TODO: deprecated use dependency injection.
```
Comment (ribel) [inline src/Plugin/KPIDatasource/DrupalKPIDatasource.php:23]: @dmytriikaiun I think we can remove this TODO now, since we started to use DI?

**Review by ribel — CHANGES_REQUESTED** [review]
@dmytriikaiun, In general, all looks good to me, just have few minor questions.

File: assets/js/morris-chart.js (hunk 1 above)
Comment (dmytriikaiun) [reply assets/js/morris-chart.js:7]: @ribel it fix this issue -> https://getopensocial.atlassian.net/browse/YANG-4023
Sometimes `context`  not correct.

File: src/Plugin/Block/KPIBlockContentBlock.php (hunk 2 above)
Comment (dmytriikaiun) [reply src/Plugin/Block/KPIBlockContentBlock.php:64]: Ok

File: src/Plugin/KPIDatasource/DrupalKPIDatasource.php (hunk 3 above)
Comment (dmytriikaiun) [reply src/Plugin/KPIDatasource/DrupalKPIDatasource.php:23]: Yes, I guess we can

**Review by ribel — APPROVED** [review]

**Pull request merged**